x/evm/keeper/grpc_query.go

```diff
@@ -0,0 +1,28 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/axelarnetwork/axelar-core/x/evm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var _ types.QueryServiceServer = baseKeeper{}
+
+func (k baseKeeper) BurnerInfo(c context.Context, req *types.BurnerInfoRequest) (*types.BurnerInfoResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	if !k.HasChain(ctx, req.Chain) {
+		return nil, sdkerrors.Wrapf(types.ErrEVM, "unkown chain '%s'", req.Chain)
+	}
+
+	burnerInfo := k.ForChain(req.Chain).GetBurnerInfo(ctx, common.HexToAddress(req.Address))
+	if burnerInfo == nil {
+		return nil, sdkerrors.Wrap(types.ErrEVM, fmt.Sprintf("unknown burner address '%s'", req.Address))
```

Comment:
Can we define an explicit error code here for "unknown burner address"?

The microservices mainly wants to know whether or not a given address is a burner, so to determine this it will have to do string matching.

String matching to resolve errors has caused problems on the client side.

Alternatively we could have an additional `isBurnerAddress` or `burnerAddressExists` query which returns a boolean?

Reply:
added error code

Reply:
i am seeing a strange behaviour with error codes on the grpc query service:

```
// module errors
var (
	// Code 1 is a reserved code for internal errors and should not be used for anything else
	_                         = sdkerrors.Register(ModuleName, 1, "internal error")
	ErrEVM                    = sdkerrors.Register(ModuleName, 2, "bridge error")
	ErrRotationInProgress     = sdkerrors.Register(ModuleName, 3, "key rotation in progress")
	ErrSignCommandsInProgress = sdkerrors.Register(ModuleName, 4, "signing for command batch in progress")
	ErrBurnerInfoNotFound     = sdkerrors.Register(ModuleName, 5, "burner info not found")
)
```

Yet, if the query fails, the error code is not 5:

```
{
  "code": 3,
  "message": "unkown chain '': burner info not found: invalid request",
  "details": [
  ]
}
```

the label is indeed the expected one, but not the code. Is this expected behavior?

Reply:
replaced this custom error code by `NotFound` provided by `google.golang.org/grpc/codes`. Which according to the docs, thats is how we are supposed to provide error codes: https://docs.cosmos.network/master/building-modules/query-services.html

Reply:
Was the error code defined as a proto enum?

Reply:
Sure this will work for this use case. Think we should revisit the error codes in a coming version though as there are several instances where more granular codes would be useful.

Reply:
Nop, the error code is only defined as a constant value. If this will do for now, please approve the PR so that we can merge it